tests/test_spark.py

```diff
@@ -430,3 +433,75 @@ def test_task_stdout():
         assert stdout_file["size"] > 0, "stdout file should have content"
     finally:
         sdk_install.uninstall(utils.SPARK_PACKAGE_NAME, service_name)
+
+
+@pytest.mark.sanity
+def test_handling_wrong_request_to_spark_dispatcher():
+    service_name = utils.FOLDERED_SPARK_SERVICE_NAME
+    utils.require_spark(service_name=service_name)
+
+    json_data = sdk_cmd.run_cli("dcos task --json")
+    dispatcher_ip = json_data['statuses'][0]['container_status']['network_infos'][0]['ip_addresses'][0]['ip_address']
+    port = json_data['discovery']['ports'][0]['number']
+
+    headers = {
+        'Content-Type': 'application/json;charset=UTF-8',
+    }
+    data = {"action": "CreateSubmissionRequest",
+            "clientSparkVersion": "2.3.2",
+            "appResource": "noopTest.jar",
+            "mainClass": "noop.Test",
+            "appArgs": ["arg1", "arg2"],
+            "environmentVariables": {
+                "PATH": "/dev/null"
+            },
+
+            "sparkProperties": {
+                "spark.app.name": "TestDispatcher",
+
+            },
+
+            }
+    response = requests.post('http://{0}:{1}/submissions/create'.format(dispatcher_ip, port), headers=headers, data=data)
```

Comment:
`requests.post` will execute a HTTP request from the machine where the test is running. Given that it is not guaranteed that Spark Dispatcher will be running on public IP, this request will fail for private IP address. In order to overcome this, the request should be executed _from within_ DCOS cluster, e.g. by using `node ssh "curl..."` or similar approach.

Reply:
We have handy ways to run cmds on masters/agents, see: https://github.com/mesosphere/spark-build/blob/master/testing/sdk_cmd.py#L268